move hash to templ

hgouraud opened the pull request:
fixes #2280

hgouraud committed:
move hash to templ

a2line commented:
I don't get why we have a ocaml5.3 CI failure popping now because of a conflict Ocaml version from [gen_js_api](https://ocaml.org/p/gen_js_api/latest#dependencies). All commits pushed to master passed 3 hours ago…!?

Edit 2 days later: It seems it was a temporary error… 5.3.0 builds are now passing!

a2line committed:
desactivate parents navigation button for dag

fix #2278

a2line committed:
use long form for hashfile apply calls

a2line commented:
@hgouraud can you confirm it's working with a non-empty value for `%etc_prefix;`?

hgouraud commented:
no, it is not working!
in normal mode, I have
` link rel="stylesheet" href="css/css.css?hash=de7ebf333e81605a294d2ebba9e1e54d"`
in CGI mode, I have
`link rel="stylesheet" href="../distribution/gw/etc/css/css.css?hash="`
Faut creuser un peu du côté resolve_asset_path en mode CGI, mais j'ai mis une trace dans hash_file_cached, mais je ne vois pas cette trace!!

hgouraud commented:
```
            Printf.eprintf "Hash 1: %s\n" file;
            let fpath = Util.resolve_asset_file conf file in
            Printf.eprintf "Hash 2: %s\n" fpath;
```
donne
```
Hash 1: ../distribution/gw/etc/css/all.min.css
Hash 2: etc/../distribution/gw/etc/css/all.min.css
```

a2line commented:
So %etc_prefix; is working.. it's another problem here!

a2line committed:
don't resolve assets, use only file for cgi

hgouraud committed:
unbound error log syntax

a2line committed:
fix template resolution regression with wildcard support

The recent refactoring of etc_file_name broke template resolution when
using "template=*" in the configuration file. The "*" character has
special meaning: it allows any template to be used, not just those
explicitly listed.

Issues fixed:
- Restore support for "*" wildcard in template configuration
- Fix resolve_asset_file to use the same template search logic as
  etc_file_name (fixes CSS/JS file resolution in template subdirectories)
- Templates specified via "?templ=templm" now work correctly again

Changes:
- Filter "*" from template list while preserving its "allow all" semantic
- Introduce find_template_file as a generic function for all file types
- etc_file_name now calls find_template_file with auto_txt=true (historical compatibility)
- resolve_asset_file now calls find_template_file with auto_txt=false
- Simplify code using List.concat_map and List.nth_opt
- Restore explanatory comments about search order

The search order remains:
1. bases/etc/mybase/templx/  (current template in base)
2. bases/etc/mybase/         (base root)
3. gw/etc/templx/            (current template in assets)
4. gw/etc/                   (assets root)

This refactoring eliminates code duplication while maintaining backward
compatibility for template files that expect automatic .txt extension.

a2line committed:
send HTTP headers before template content

* `Content-Type` need to be sent first in CGI.
* New `Util.is_full_html_template` function testing `<!DOCTYPE html>` presence
in the template file to check if it’s a full HTML page that needs HTTP header.
* Reduce updateInd/Fam.ml `include_perso_header` to include_menubar only to avoid
duplicate HTML headers in the body that is structuraly useless and false.

Fixes #2285

a2line committed:
add unbreakable space before circa

a2line committed:
remove superfluous dot

a2line committed:
missing space

a2line committed:
larger modal + vertical align

a2line commented:
@hgouraud what about what's left in a343db7b4770fb17aafb49149b81d5503662731c after rebase behind merge of #2288?
